### PR TITLE
Merge remote-tracking branch 'refs/remotes/ugrid-conventions/master'

### PR DESCRIPTION
**Pull request by GLang-BAW**
To my understanding the so far used attributes flag_values and flag_meanings in variable Mesh2_face_links were not consistent with CF conventions: "The attributes flag_values and flag_meanings are intended to make variables that contain flag values self-describing. The flag_values and flag_meanings attributes describe a status flag consisting of mutually exclusive coded values." 

From this one can conclude:

1. Each data item of the variable is interpreted in terms of the available flag_values and flag_meanings.

2. For each valid data item there must be an entry in the list of flag_values and flag_meanings. Otherwise we have no meaning for a specific data item.

It is herewidth suggested to use _FillValue to indicate missing neighbors in Mesh2_face_links.

### CODE REVIEW

**Comment by GLang-BAW** [comment]
Sorry. The first time somehow the changes were not transferred. 

Günther

**Comment by rsignell-usgs** [comment]
@GLang-BAW , I like this suggestion.   Any opposition?
@JonathanGregory, this is an appropriate use of `_FillValue`, right?

**Comment by JonathanGregory** [comment]
Do you mean you would put `_FillValue` in the `flag_values`? I suppose that would be legal - I can't see why not. However, I think it would not normally be done. Perhaps this should be clarified in the convention, but I guess that if a data value is `_FillValue`, it's assumed to be missing data, and therefore not looked up in `flag_values`. If you want to indicate more precisely that this is a non-existing neighbour (rather than some other reason for missing data), perhaps it would be better to define a particular value for that?

**Comment by ChrisBarker-NOAA** [comment]
@JonathanGregory: no, I think the idea is to simply use _FillValue, rather than flag_values at all.

In this case, these are indices to other parts of the mesh, I think there is only one reason to not have a valid index: whatever you would be referencing doesn't exist.

In the whole standard, there are multiple reasons that it doesn't exist:
  - the cell is on a boundary, and thus has no neighbor on that side
  - a cell has fewer vertices than the max allowed.
  - any others?

But in a given context, I think only one meaning (or maybe "nothing valid here" is fine, even if there is more than one reason for not valid.

The only trick I see is that netcdf has defaults for _FillValue, and in this case, folks would almost never want to use that default. (-1 is a common convention).

**Comment by JonathanGregory** [comment]
I see. I agree it would be OK to use `_FillValue`, if there is no case where you want to distinguish between "this datum should exist but is missing" and "this datum really should not exist". Jonathan

**Comment by ChrisBarker-NOAA** [comment]
Thanks Jonathan. And I think that distinction isn't needed here.

So I say we go with this -- but I haven't actually read the new text yet :-)

Someone needs to review the text, and we need a couple more OKs before we merge.

**Comment by rsignell-usgs** [comment]
This is more than okay, it corrects a mistake!   Merging...

**Comment by ocefpaf** [comment]
> This is more than okay, it corrects a mistake! Merging...

Than a version number bump would be in order, no? I will update the page once you think it is OK to do so.

File: src/index.md [hunk 1]
```diff
@@ -192,10 +192,10 @@ Mesh2_face_edges:long_name = "Maps every triangular face to its three edges." ;
 Mesh2_face_edges:start_index = 1 ;
 integer Mesh2_face_links(nMesh2_face, Three) ;
 Mesh2_face_links:cf_role = "face_face_connectivity" ;
-Mesh2_face_links:long_name = "Indicates which other faces neighbor each face." ;
+Mesh2_face_links:long_name = "neighbor faces for faces" ;
 Mesh2_face_links:start_index = 1 ;
-Mesh2_face_links:flag_values = -1 ;
-Mesh2_face_links:flag_meanings = "out_of_mesh" ;
+Mesh2_face_links:_FillValue = -999 ;
```
Comment (ChrisBarker-NOAA) [inline src/index.md]: This is correct, of course, but I think the example should use -1 as the _FillValue -- that's in common usage.

**Comment by ChrisBarker-NOAA** [comment]
yes, I think a (minor) version number bump is in order -- maybe change the example to -1 at the same time?

numbers like -9999 are commonly used as "missing values" in CSV Files, etc, but I"ve ony seen -1 used in index arrays like this case.

Frankly, I'm likely to write code that checks for < 0 anyway, and any index less than 0 is invalid.

-CHB

**Comment by ChrisBarker-NOAA** [comment]
we're at 1.0 now -- so should this be 1.1 or 1.0.1 ? this is a pretty minor change.

**Pull request merged**